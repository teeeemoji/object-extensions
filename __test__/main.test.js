/**
 * Summary.
 *  main test file.
 *
 * Description
 *  This is the main test file
 *
 * @file   This main test file for class object-property-extensions.
 * @author teeeemoji.
 * @since 2019-10-21
 */
import {hasOwnProperty, safeGet, safeSet} from '../index'

const TEST_OBJ = {
  a: null,
  one: {
    two: {
      three: 4
    }
  },
  arr: [
    {
      value: 2
    }
  ]
}

describe('hasOwnProperty', () => {
  let testObj
  beforeEach(() => {
    testObj = Object.assign({}, TEST_OBJ)
  })

  test('hasOwnProperty one.two.three', () => {
    expect(hasOwnProperty(testObj, 'one.two.three')).toMatchSnapshot()
  })
  test('not hasOwnProperty one.two.three.four', () => {
    expect(hasOwnProperty(testObj, 'arr.1.value')).toMatchSnapshot()
  })
})
describe('safeGet', () => {
  let testObj
  beforeEach(() => {
    testObj = Object.assign({}, TEST_OBJ)
  })

  test('safeGet one.two.three', () => {
    expect(safeGet(testObj, 'one.two.three')).toMatchSnapshot()
  })
  test('safeGet arr.0.value', () => {
    expect(safeGet(testObj, 'arr.0.value')).toMatchSnapshot()
  })
  test('safeGet arr.1.value', () => {
    expect(safeGet(testObj, 'arr.1.three.four')).toMatchSnapshot()
  })
})
describe('safeSet', () => {
  const newVal = 1
  let testObj
  beforeEach(() => {
    testObj = Object.assign({}, TEST_OBJ)
  })

  test('safeSet one.two.three', () => {
    expect(safeSet(testObj, 'one.two.three', newVal)).toMatchSnapshot()
    expect(testObj.one.two.three).toBe(newVal)
  })
  test('safeSet arr.0.value', () => {
    expect(safeSet(testObj, 'arr.0.value', newVal)).toMatchSnapshot()
    expect(testObj.arr[0].value).toBe(newVal)
  })
  test('safeSet arr.1.value', () => {
    expect(safeSet(testObj, 'arr.1.value')).toMatchSnapshot()
  })
})
