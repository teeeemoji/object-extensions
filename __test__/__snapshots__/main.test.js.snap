// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hasOwnProperty hasOwnProperty one.two.three 1`] = `true`;

exports[`hasOwnProperty not hasOwnProperty one.two.three.four 1`] = `false`;

exports[`safeGet safeGet arr.0.value 1`] = `2`;

exports[`safeGet safeGet arr.1.value 1`] = `undefined`;

exports[`safeGet safeGet one.two.three 1`] = `4`;

exports[`safeSet safeSet arr.0.value 1`] = `true`;

exports[`safeSet safeSet arr.1.value 1`] = `false`;

exports[`safeSet safeSet one.two.three 1`] = `true`;
